scripts/utils/trading_strategy_helpers.js

```diff
@@ -360,6 +361,67 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
+   * the resulting transaction is that of transfering all specified funds from master through its brackets
+   * followed by approval and deposit of those same tokens into BatchExchange on behalf of each bracket.
+   *
+   * @param {string} masterAddress Ethereum address of Master Gnosis Safe (Multi-Sig)
+   * @param {Transfer[]} transferList List of {@link Deposit} that are to be bundled together
+   * @param {boolean} [useWei=false] flag allowing user to specify if amounts are already in Base token units
+   * @param {boolean} [unsafe=false] does not perform balance verification
+   * @param {boolean} [debug=false] prints log statements when true
+   * @returns {Transaction} all the relevant transaction information used for submission to a Gnosis Safe Multi-Sig
+   */
+  const buildTransferDataFromList = async function (masterAddress, transferList, useWei = false, unsafe = false, debug = false) {
+    const log = debug ? (...a) => console.log(...a) : () => {}
+
+    const uniqueTokens = uniqueItems(transferList.map((t) => t.tokenAddress))
+    const tokenInfo = fetchTokenInfoForFlux(transferList)
+
+    // Will be used to make sufficient balance assertion before transfer.
+    const cumulativeAmounts = new Map(uniqueTokens.map((address) => [address, new BN(0)]))
+    const transactions = await Promise.all(
+      transferList.map(async (transfer) => {
+        const token = await tokenInfo[transfer.tokenAddress]
+        let unitAmount = new BN(transfer.amount)
+        if (!useWei) {
+          unitAmount = toErc20Units(transfer.amount, token.decimals)
+        }
+        // Accumulate amounts being transfered for each token.
+        cumulativeAmounts.set(token.address, cumulativeAmounts.get(token.address).add(unitAmount))
+
+        log(`Transferring ${unitAmount} ${token.symbol} to ${transfer.receiver} from ${shortenedAddress(masterAddress)}`)
+        const transferData = token.instance.contract.methods.transfer(transfer.receiver, unitAmount.toString()).encodeABI()
+        return {
+          operation: CALL,
+          to: token.address,
+          value: 0,
+          data: transferData,
+        }
+      })
+    )
+    log(`Transfer bundle contains ${transactions.length} elements and sends ${uniqueTokens.length} distinct tokens.`)
+    if (!unsafe) {
+      // Ensure sufficient funds.
+      await Promise.all(
+        uniqueTokens.map(async (tokenAddress) => {
+          const token = await tokenInfo[tokenAddress]
+          const masterBalance = await token.instance.balanceOf(masterAddress)
+          if (masterBalance.lt(cumulativeAmounts.get(tokenAddress))) {
+            throw new Error(
+              `Master Safe has insufficient ${token.symbol} balance (${masterBalance.toString()} < ${cumulativeAmounts
+                .get(tokenAddress)
+                .toString()})`
+            )
+          }
+        })
+      )
+      log("Balance verification passed")
+    }
+    return buildBundledTransaction(transactions)
```

Comment:
Great function here the `buildBundledTransaction`

Just an idea, if you are interested for future versions :) 
`execTransactionData` does 2 things:
- Prepare the multisend call, with all the encoded smaller calls
- Send the transaction

If we just remove the part of sending the transaction (or move it to somewhere else), we could for example print the details of the transaction. This would allow anyone to execute this transaction WITHOUT any private key. Just signing it with his Ledger or Gnosis Safe in MyEtherWallet

Reply:
We do not want to mix send transaction with this other thing because we still need a way to prompt the user after we have acquired all the transaction details. "This is what you are about to sign would are you sure?"

Reply:
I mean this for running it without the need of a PK, not sure if I explained myself very well. 

It would be a flag, that would, instead of sending the transaction, would print the `to, data, gas` so anyone can sign that MANUALLY in myetherwallet.com

This would remove the need of a `proposer` account.
I don't mean to include it in this PR. Just consider for the project, since it can make sense also for the CMM.

Happy to discuss it if personally too

Reply:
As it works right now `buildBundledTransaction` is very generic, in the sense that it's not the part of the code that requires the proposer. Anything can be bundled together and there is no need for a private key.
Note also that the functions `execTransaction` and `execTransactionData` are only used for testing. (We keep them in the main project in case we wanted to bypass the web interface in the future.) The actual transaction that is submitted to the blockchain is currently not built by us but by the Multisig web interface. So, it works like this:
1. we build a transaction that we would like to have executed on our master Safe (`fundAccount` here), "execute" means "call the Safe contract function `execTransaction`"
2. we tell the web interface "put up this transaction to be executed". In order to do this, we need a signature of at least one user, this is where the need of a proposer account come from
3. the web interface collects user signatures and builds the actual transaction that will be submitted to the blockchain from our proposed transaction and the signatures
4. someone submits the web interface's transaction to the blockchain from the web interface.

Our script is directly involved only in step 1. So, I don't really know what myetherwallet.com does, but I believe the functions here are already fairly general so they might already be working as you wish they would be.
Anyway, happy to have more eyes looking at this project, also happy to discuss if you want more details/have more ideas!